addons/mrp/models/mrp_production.py

```diff
@@ -291,6 +295,8 @@ def _compute_components_availability(self):
             elif forecast_date:
                 production.components_availability = _('Exp %s', format_date(self.env, forecast_date))
                 production.components_availability_state = 'late' if forecast_date > production.date_planned_start else 'expected'
+            if not production.move_raw_ids:
```

Comment:
I would move this condition above the declaration of `forecast_cast` and `continue` if no move_raw_ids.

Reply:
Makes sense, done.